### ISSUE TITLE
[BUG] Too high `package:meta` dependency constraint

### ISSUE THREAD

**Issue by Albert221** [fleaflet/flutter_map#1896]
### What is the bug?

```
Because every version of flutter_test from sdk depends on meta 1.11.0 and flutter_map >=7.0.0 depends on meta ^1.12.0, flutter_test from sdk is incompatible with flutter_map >=7.0.0.
So, because <package> depends on both flutter_map ^7.0.0 and flutter_test from sdk, version solving failed.
```

I'm on Flutter 3.19.5 for now and can't update yet, but the package's pubspec says minimum Flutter 3.10 is needed, so maybe we can make the constraint for `meta` a little more loose?

### How can we reproduce it?

Create new Flutter project with Flutter 3.19.* and then `flutter pub add flutter_map:7.0.0`

```
$ flutter pub add flutter_map:7.0.0
Note: meta is pinned to version 1.11.0 by flutter_test from the flutter SDK.
See https://dart.dev/go/sdk-version-pinning for details.


Because every version of flutter_test from sdk depends on meta 1.11.0 and flutter_map >=7.0.0 depends on meta ^1.12.0,
  flutter_test from sdk is incompatible with flutter_map >=7.0.0.
So, because testtt depends on both flutter_map 7.0.0 and flutter_test from sdk, version solving failed.
```

### Do you have a potential solution?

Make the constraint for `meta` something like `>=1.11.0 <1.13.0` instead

### Platforms

Flutter 3.19

### Severity

Erroneous: Prevents normal functioning and causes errors in the console

**Comment by RuoLiDevelop** [comment]
file` pubspec.yaml`
```
dependency_overrides:
  meta: ">=1.11.0 <1.15.0"
```

**Comment by Albert221** [comment]
@RuoLiDevelop that's a workaround, yes, but I'm opting for a solution

**Comment by monsieurtanuki** [comment]
Probably fixed by #1897

**Comment by JaffaKetchup** [comment]
Hi @amin79,
We're working on a couple more bug fixes, then we should be able to release. For the time being, please use this (https://github.com/fleaflet/flutter_map/issues/1896#issuecomment-2137161605) workaround.
